Automattic/jetpack
# CRM: OAuth/Gmail issue stripes HTML code, including links and images

Fixes https://github.com/Automattic/zero-bs-crm/issues/3109

## Proposed changes:
This PR fixes an issue when users use the OAuth/Gmail method to send emails. It didn't send the plain/text part, so Gmail created a new one based on the HTML, which implies stripping HTML elements (i.e. links and images). But after debugging it in depth, the transfer encoding was also incorrect (`Content-Transfer-Encoding: quoted-printable`).

I added a new boundary (`Content-Type: multipart/mixed`) structure to allow attachments and plain/text content (`Content-Type: multipart/alternative`):

- multipart/mixed
   - multipart/alternative => for the plain and the HTML content
   - the attachments part

### Other information:


## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
N/A

## Testing instructions:
- Apply this patch
- Set up an OAuth in Google and add the credentials to your testing site. If you are testing it locally, be sure you add the localhost URI, or if you use testing local domain, should be a `.com` domain (pointing to 127.0.0.1 in your hosts file). But better if you test it in an external testing site accessible because the Gmail image cache needs it to show you any image or logo.
- Configure OAuth/Gmail as email delivery system in the JPCRM Mail setting
- Create a mail campaign with MC with a small audience (less than 3 to not saturate your inbox). At least one contact email should be accessible to be able to check the email.
- Select the template with logo. Ensure you've set up one in the business information (JPCRM settings)
- Add any link
- Send the test to any valid email. You should able to see the unsubscribe link
- Send the broadcast
- Check that you receive the email, with the links you added, also you will see the company logo.
- **Bonus:** Because of I touched the email content at a low level using multiple boundaries, test this: Create an invoice and send it via email (valid email) as a PDF attachment. Check that you receive the email correctly with the attachment.
